pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -203,7 +213,29 @@ func init() {
 }
 
 // Initialize passes a Kubernetes clientBuilder interface to the cloud provider
-func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {}
+func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {
+	if vs.cfg == nil {
+		return
+	}
+
+	// Only on controller node it is required to register listeners.
+	var err error
+	vs.clientBuilder = clientBuilder
+	vs.client, err = clientBuilder.Client("vSphere-cloud-provider")
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to create kubeclient: %v", err))
+	}
+
+	// Register callbacks for node updates
+	factory := informers.NewSharedInformerFactory(vs.client, 5*time.Minute)
```

Comment:
Could you explain what is this 5 mins for?

Reply:
Its the default resync time. Looks like NodeInformer cache is refreshed after this much of time.